trace/dawr.py

```diff
@@ -0,0 +1,135 @@
+        shutil.copyfile(self.get_data('wp1.c'),
+                        os.path.join(self.teststmpdir, 'wp1.c'))
+        shutil.copyfile(self.get_data('wp2.c'),
+                        os.path.join(self.teststmpdir, 'wp2.c'))
+        shutil.copyfile(self.get_data('wp3.c'),
+                        os.path.join(self.teststmpdir, 'wp3.c'))
+        shutil.copyfile(self.get_data('Makefile'),
+                        os.path.join(self.teststmpdir, 'Makefile'))
+        build.make(self.teststmpdir)
+        os.chdir(self.teststmpdir)
+        self.temp_file = tempfile.NamedTemporaryFile().name
+
+    def test_wp1(self):
+        """
+        Setting Read/Write watchpoint on single variable using awatch and
+        executing the program
+        """
+        child=pexpect.spawn('gdb ./wp1',encoding='utf-8')
+        time.sleep(0.3)
+        child.logfile = sys.stdout
+        child.expect('(gdb)')
+        return_value=[]
+        child.sendline('awatch a')
+        return_value.append(child.expect_exact(['Hardware access (read/write) \
+                watchpoint 1: a',pexpect.TIMEOUT]))
+        child.sendline('r')
+        return_value.append(child.expect_exact(['Value = 10',pexpect.TIMEOUT]))
+        child.sendline('c')
+        return_value.append(child.expect_exact(['New value = 20',pexpect.TIMEOUT]))
+        print(return_value)
+        for i in return_value:
+            if i!=0:
+                self.fail('Test case failed for 1 variable')
+
+    def test_wp2(self):
+        """
+        Setting Read/Write watchpoints on two variables using awatch and
+        executing the program
+        """
+        child=pexpect.spawn('gdb ./wp2',encoding='utf-8')
+        time.sleep(0.3)
+        child.logfile = sys.stdout
+        child.expect('(gdb)')
+        return_value=[]
+        child.sendline('awatch a')
+        return_value.append(child.expect_exact(['Hardware access (read/write) \
+                watchpoint 1: a',pexpect.TIMEOUT]))
+        child.sendline('awatch b')
+        return_value.append(child.expect_exact(['Hardware access (read/write) \
+                watchpoint 2: b',pexpect.TIMEOUT]))
+        child.sendline('r')
+        return_value.append(child.expect_exact(['Value = 10',pexpect.TIMEOUT]))
+        child.sendline('c')
+        return_value.append(child.expect_exact(['New value = 20',pexpect.TIMEOUT]))
+        child.sendline('c')
+        return_value.append(child.expect_exact(['Value = 10',pexpect.TIMEOUT]))
+        child.sendline('c')
+        return_value.append(child.expect_exact(['New value = 20',pexpect.TIMEOUT]))
+        child.sendline('c')
+        return_value.append(child.expect_exact(['Value = 20',pexpect.TIMEOUT]))
+        child.sendline('c')
+        return_value.append(child.expect_exact(['Value = 20',pexpect.TIMEOUT]))
+        child.sendline('c')
+        return_value.append(child.expect_exact(['exited normally',pexpect.TIMEOUT]))
```

Comment:
@akanksha216 see if it is possible to optimized similar code lines in a loop

for all values in a list[] :
     return_value.append(child.expect_exact(['New value = ',pexpect.TIMEOUT]))
        child.sendline('c')